src/ngx_http_lua_util.c

```diff
@@ -3980,4 +3985,84 @@ ngx_http_lua_get_raw_phase_context(lua_State *L)
     return 1;
 }
 
+
+ngx_http_cleanup_t *
+ngx_http_lua_cleanup_add(ngx_http_request_t *r, size_t size)
+{
+    ngx_http_cleanup_t  *cln;
+    ngx_http_lua_ctx_t  *ctx;
+
+    r = r->main;
```

Comment:
I don't think this line is needed here and actually this may cause problems since the main request may not use any Lua at all, in which case the ngx_lua ctx for the main request is always NULL and we lose cleanup record recycling in the suberquest using Lua. The same applies to the `ngx_http_lua_cleanup_free` function below.